manage.py

```diff
@@ -373,25 +294,65 @@ def update_all(processes=1):
     """
     logger.info('Refreshing materialized views...')
-    execute_sql_file('data/refresh_materialized_views.sql')
+    materialized_view_names = {
+        'filing_amendments_house_senate': ['ofec_house_senate_paper_amendments_mv'],
+        'sched_f': ['ofec_sched_f_mv'],
+        'reports_ie': ['ofec_reports_ie_only_mv'],  # Anomale
+        'electioneering_by_candidate': ['ofec_electioneering_aggregate_candidate_mv'],
+        'candidate_history': ['ofec_candidate_history_mv'],
+        'candidate_detail': ['ofec_candidate_detail_mv'],
+        'candidate_election': ['ofec_candidate_election_mv'],
+        'candidate_history_latest': ['ofec_candidate_history_latest_mv'],
+        'omnibus_dates': ['ofec_omnibus_dates_mv'],
+        'election_outcome': ['ofec_election_result_mv'],
+        'sched_e_by_candidate': ['ofec_sched_e_aggregate_candidate_mv'],
+        'filing_amendments_presidential': ['ofec_presidential_paper_amendments_mv'],
+        'filing_amendments_pac_party': ['ofec_pac_party_paper_amendments_mv'],
+        'cand_cmte_linkage': ['ofec_cand_cmte_linkage_mv'],
+        'communication_cost_by_candidate': ['ofec_communication_cost_aggregate_candidate_mv'],
+        'electioneering': ['ofec_electioneering_mv'],
+        'filing_amendments_all': ['ofec_amendments_mv'],
+        'reports_house_senate': ['ofec_reports_house_senate_mv'],
+        'reports_pac_party': ['ofec_reports_pacs_parties_mv'],
+        'reports_presidential': ['ofec_reports_presidential_mv'],
+        'committee_history': ['ofec_committee_history_mv'],
+        'communication_cost': ['ofec_communication_cost_mv'],
+        'filings': ['ofec_filings_amendments_all_mv', 'ofec_filings_mv'],
+        'totals_combined': ['ofec_totals_combined_mv'],
+        'totals_ie': ['ofec_totals_ie_only_mv'],
+        'totals_house_senate': ['ofec_totals_house_senate_mv'],
+        'totals_presidential': ['ofec_totals_presidential_mv'],
+        'candidate_aggregates': ['ofec_candidate_totals_mv'],
+        'candidate_flags': ['ofec_candidate_flag'],
+        'sched_c': ['ofec_sched_c_mv'],
+        'sched_a_by_state_recipient_totals': ['ofec_sched_a_aggregate_state_recipient_totals_mv'],
+        'rad_analyst': ['ofec_rad_mv'],
+        'committee_detail': ['ofec_committee_detail_mv'],
+        'committee_fulltext': ['ofec_committee_fulltext_mv'],
+        'sched_a_by_size_merged': ['ofec_sched_a_aggregate_size_merged_mv'],
+        'totals_pac_party': ['ofec_totals_pacs_parties_mv', 'ofec_totals_pacs_mv', 'ofec_totals_parties_mv'],
+        'large_aggregates': ['ofec_entity_chart_mv'],
+        'candidate_fulltext': ['ofec_candidate_fulltext_mv'],
+        'totals_candidate_committee': ['ofec_totals_candidate_committees_mv']
+    }
+    graph = flow.get_graph()
+    for node in nx.topological_sort(graph):
+        for mv in materialized_view_names[node]:
+            logger.info('Refreshing %s', mv)
+            db.session.execute("REFRESH MATERIALIZED VIEW %s" % mv)
```

Comment:
One of the issues we ran into in the past was that due to some of the dependencies between the materialized views and tables (one of the Schedule A aggregates in particular), just calling `REFRESH MATERIALIZED VIEW` wasn't enough, hence the need to go through and recreate all of them and do the rename.

Will this work correctly given that they are being refreshed in the correct order?  Do we need to worry about the indexes at all and perform a reindex on them?

I definitely like that this is much simpler and leveraging the database itself more, but just want to be sure it'll work as we need it to!

Reply:
I think that this code ensures that the materialized views are refreshed in the correct order. We should test it on `dev` to make sure.

Reply:
I have tested it on `dev` and it works correctly. I will be changing the `REFRESH MATERIALIZED VIEW` to add `CONCURRENTLY` for lesser interference with other `SELECT` queries.